enhancements/microshift/disabling-storage-components.md

```diff
@@ -0,0 +1,250 @@
+outcomes. Unit tests will be written as necessary.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- N/A
+
+### Tech Preview -> GA
+
+- N/A
+
+### GA
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+## Upgrade / Downgrade Strategy
+
+MicroShift gracefully handles config fields it does not recognize. Downgrading from a system with these config fields
+will not break MicroShift start up.
+
+For clusters that disabled the storage components, a downgrade would result in these components being deployed. This
+will not break the user's cluster but will result in additional resource overhead. This situation can't be avoided, as
+even if the user scaled down the storage component replica sets, restarting or rebooting microshift would cause the
+resources to be re-applied, thus scaling the sets back up.
+
+LVMS installation automatically reversible so upgrades and downgrades do not present a danger to user's data.
+Upgrading a cluster with LVMS to a cluster with LVMS set to disabled will not cause the storage components to be deleted.
+
+## Version Skew Strategy
+
+LVMS is versioned separately from MicroShift. However, MicroShift pins the version of LVMS during rebasing. This creates
+a loose coupling between the two projects, but does not pose a risk of incompatibility. LVMS runs in cluster workloads
+and thus does not directly interact with MicroShift internals.
+
+LVMS does not interact with the MicroShift config API, so it will not be affected by this change.
+
+## Operational Aspects of API Extensions
+
+With LVMS and the CSI components deployed, the cluster overhead is increased by roughly 50Mb and 0.1% CPU. Though
+comparatively small by OCP standards, 50Mb of memory is a non-trivial amount on far-edge devices.
+
+## Support Procedures
+
+- N/A
+
+## Alternatives
+
+Do Nothing. Continue to use the MicroShift or LVMD config to determine deployment of LVMS and/or snapshotting. Directing
+the user to leverage low-level processes to achieve this goal is an anti-pattern and would provide a poor UX.
+
+Install LVMS and CSI via rpm, as is done with other cluster components. MicroShift is strongly opinionated towards LVMS,
+with significant logic for managing some of the LVMS life cycle and default configs. If LVMS manifests and container
+images were installed via an RPM, the logic for handling dynamic defaults would need to be implemented elsewhere. Either as a %post-install stage or as a separate system service. Neither of these approaches is suitable for the goal of this EP, 
```

Comment:
Another 2 cents, just throwing ideas: I think this could be a sidecar container with the logic Jakob implemented (watch /etc/microshift and feed /var/lib/microshift/lvms). Maybe something for distant future when we divorce lvms and microshift